models/ldndc/R/met2model.LDNDC.R

```diff
@@ -0,0 +1,247 @@
+        simu_days <- 1:PEcAn.utils::days_in_year(year)
+        
+        # Whole year, from first to last index
+        ind_s <- 1
+        ind_e <- which.max(sec)
+        
+      }else if(year == start_year & year != end_year){
+        
+        # Simulation days
+        simu_days <- seq(lubridate::yday(start_date),PEcAn.utils::days_in_year(year))
+        
+        # Check the function to see how the start index is calculated
+        ind_s <- start_index(units, start_date, sec)
+        ind_e <- which.max(sec)
+        
+      }else if(year != start_year & year == end_year){
+        
+        # Simulation days
+        simu_days <- seq(1:lubridate::yday(end_date))
+        
+        # Check the function to see how the end index is calculated
+        ind_s <- 1
+        ind_e <- end_index(units, start_date, end_date, sec, tstep)
+        
+      }else{
+        
+        # Simulation days
+        simu_days <- seq(lubridate::yday(start_date), lubridate::yday(end_date))
+        
+        # Need to calculate both first and last index by using functions
+        ind_s <- start_index(units, start_date, sec)
+        ind_e <- end_index(units, start_date, end_date, sec, tstep)
+        
+      }
+      
+      # Based on the simulation days, this is year, day and subday info
+      # for model to use.
+      y <- rep(year, length(simu_days)*tstep) #year
+      d <- rep(simu_days, each = tstep) #day
+      subd <- rep(1:tstep, length(simu_days)) #subday
+      
+      # On above the starting and ending indexes has been determined.
+      # Here those are just collapsed.
+      ind <- ind_s:ind_e
+      
+        
+      
+      ## Info for climate file that model is using ##
+      # Latitude and longitude
+      lat <- ncdf4::ncvar_get(nc, "latitude")
+      lon <- ncdf4::ncvar_get(nc, "longitude")
+      
+      
+      # Average air temperature
+      Tair <-ncdf4::ncvar_get(nc, "air_temperature")[ind]  ## in Kelvin
+      tavg <- udunits2::ud.convert(Tair, "K", "degC")
+      
+      # Wind speed
+      wind <- try(ncdf4::ncvar_get(nc, "wind_speed"))[ind]
+      
+      # Precipation
+      prec <- ncdf4::ncvar_get(nc, "precipitation_flux")[ind]
+      
+      # Global radiation
+      grad <- ncdf4::ncvar_get(nc, "surface_downwelling_shortwave_flux_in_air")[ind]  ## in W/m2
+      
+      # Vapor Pressure Deficit Pa ---> kPa, need to ne divided by 1000
+      VPD <- try(ncdf4::ncvar_get(nc, "water_vapor_saturation_deficit"))[ind]  ## in Pa
+      
+      # Relative humidity (%)
+      rhum <- ncdf4::ncvar_get(nc, "relative_humidity")[ind]
+      
+      # Air pressure Pa ---> mbar, needs to divided by 100
+      press <- ncdf4::ncvar_get(nc, "air_pressure")[ind]
+      
+      
+      # Close connection after all necessary values have been fetch
+      ncdf4::nc_close(nc)
+      
+
+      # Gather the vectors to dataframe
+      data <- as.data.frame(do.call("cbind", list(y = y, d = d, s = subd, prec = prec*86400/tstep, tavg = tavg,
```

Comment:
here consider adding a comment on each variable's unit for future reference
```suggestion
      data <- as.data.frame(do.call("cbind", list(y = y, d = d, s = subd, 
      prec = prec*86400/tstep, # mm?
      tavg = tavg, # degC
      grad = grad, # W m-2
      ...
```
and so on...